api.planx.uk/admin/migrations/splitFindProperty.test.ts

```diff
@@ -0,0 +1,117 @@
+import supertest from "supertest";
+
+import { queryMock } from "../../tests/graphqlQueryMock";
+import { authHeader } from "../../tests/mockJWT";
+import app from "../../server";
+import { Flow } from "../../types";
+
+const flowDataBeforeMigration: Flow["data"] = {
+  "_root": {
+    "edges": [
+      "dzDGABKiBN",
+      "u3GrdN2UFO"
+    ]
+  },
+  "dzDGABKiBN": {
+    "type": 9
+  },
+  "u3GrdN2UFO": {
+    "type": 8,
+    "data": {
+      "color": "#EFEFEF",
+      "resetButton": true
+    }
+  }
+};
+
+const flowDataAfterMigration: Flow["data"] = {
+  "_root": {
+    "edges": [
+      "dzDGABKiBN",
+      "TESTTESTTEST",
+      "u3GrdN2UFO"
+    ]
+  },
+  "dzDGABKiBN": {
+    "type": 9
+  },
+  "u3GrdN2UFO": {
+    "type": 8,
+    "data": {
+      "color": "#EFEFEF",
+      "resetButton": true
+    }
+  },
+  "TESTTESTTEST": {
+    "type": 12,
+    "data": {
+      "title": "About the property",
+      "description": "This is the information we currently have about the property",
+      "showPropertyTypeOverride": false
+    }
+  }
+};
+
+beforeEach(() => {
+  queryMock.mockQuery({
+    name: "GetFlowAndPublishedFlowData",
+    matchOnVariables: false,
+    data: {
+      flows_by_pk: {
+        data: flowDataBeforeMigration,
+        published_flows: [
+          {
+            id: 1,
+            data: flowDataBeforeMigration
+          }
+        ]
+      }
+    },
+  });
+
+  queryMock.mockQuery({
+    name: "UpdateFlow",
+    matchOnVariables: false, // TODO mock uniqueId() and set to 'true'
```

Comment:
the "success" test is really only meaningful if I can match on variables, but that's tricky with a randomly generated new nodeId! 

i was having a hard time figuring out how to mock the uniqueId() caller - any tips?

Reply:
So something along these lines should work if you just want a single random node Id for the entire test file - 

```ts
jest.mock("nanoid-good", () => ({
  customAlphabet: jest.fn().mockImplementation(() => () => () => "randomID123"),
}));
```

The currying done by customAlphabet and uniqueId means the syntax a little weird looking here.

If you require different random ids in different tests it gets a little more tricky as Jest hoists mocks - I've found this a little troublesome in the past. This means it's not quite as simple as just declaring and reassigning a variable outside of the scope of the mock. 

Let me know if the above works or if you!

Reply:
this works, thank you!!